### PR TITLE
Resolve type when LHS of assignment is a call and RHS is a call

### PR DESCRIPTION
**Pull request by khemrajrathore**

### CODE REVIEW

**Review by tuxology — APPROVED** [review]

**Review by tuxology — APPROVED** [review]

File: src/main/scala/ai/privado/languageEngine/ruby/passes/PrivadoRubyTypeRecoveryPassGenerator.scala [hunk 1]
```diff
@@ -131,6 +121,16 @@ private class RecoverForRubyFile(
     }
   }
 
+  def isCallHeadArgumentAScopeResolutionAndIsLastArgumentInTable(c: Call): Boolean = c.argument.headOption
+    .exists(_.isCall) && c.argument.head
+    .asInstanceOf[Call]
+    .name
+    .equals("<operator>.scopeResolution") && c.argument.head
```
[the remark this answers is not in the document]
Comment (khemrajrathore) [reply src/main/scala/ai/privado/languageEngine/ruby/passes/PrivadoRubyTypeRecoveryPassGenerator.scala:128]: It doesn't

File: src/main/scala/ai/privado/languageEngine/ruby/passes/GlobalImportPass.scala [hunk 2]
```diff
@@ -37,77 +36,25 @@ import io.shiftleft.semanticcpg.language.*
 
 import scala.collection.Seq
 
-class GlobalImportPass(cpg: Cpg, packageTable: PackageTable, globalSymbolTable: SymbolTable[LocalKey])
-    extends PrivadoSimpleCpgPass(cpg) {
-
-  /*
-  lazy val modules: Set[String] =
```
Comment (tuxology) [inline src/main/scala/ai/privado/languageEngine/ruby/passes/GlobalImportPass.scala:44]: nit: lets remove this and other commented code as well

File: src/main/scala/ai/privado/languageEngine/ruby/passes/PrivadoRubyTypeRecoveryPassGenerator.scala [hunk 3]
```diff
@@ -131,17 +136,41 @@ private class RecoverForRubyFile(
...
+    .asInstanceOf[Call]
+    .argument
+    .lastOption
+    .exists(arg =>
+      symbolTable.get(arg).union(globalSymbolTable.get(arg)).exists(isCallParentScopeResolutionMatching(_, c.code))
```
Comment (tuxology) [inline src/main/scala/ai/privado/languageEngine/ruby/passes/PrivadoRubyTypeRecoveryPassGenerator.scala:158]: nit: maybe we add a comment in statements (or maybe functions) like these. Took me a while to figure this out 🤔

**Review by tuxology — APPROVED** [review]
This is quite nice. All cases seem to be covered. Just two minor nitpicks - esp. removing comments

File: src/main/scala/ai/privado/languageEngine/ruby/passes/GlobalImportPass.scala (hunk 2 above)
Comment (khemrajrathore) [reply src/main/scala/ai/privado/languageEngine/ruby/passes/GlobalImportPass.scala:44]: @tuxology This whole code is already removed 😄

**Comment by khemrajrathore** [comment]
@tuxology have addressed PR comments

File: src/main/scala/ai/privado/languageEngine/ruby/passes/GlobalImportPass.scala (hunk 2 above)
Comment (tuxology) [reply src/main/scala/ai/privado/languageEngine/ruby/passes/GlobalImportPass.scala:44]: 🤯 Need coffee ☕

**Review by tuxology — APPROVED** [review]

**Pull request merged**